packages/viz/src/lib/basemap/create-map.ts

```diff
@@ -0,0 +1,67 @@
+import { CartoMapStyle } from './CartoMapStyle';
+import { CartoBaseMapError } from '../errors/basemap-error';
+
+interface MapOptions {
+  basemap?: string;
+  view?: {
+    bearing?: number;
+    latitude?: number;
+    longitude?: number;
+    pitch?: number;
+    zoom?: number;
+  };
+  container?: HTMLElement | string;
+}
+
+const DEFAULT_OPTIONS: MapOptions = {
+  basemap: 'positron',
+  view: {
+    bearing: 0,
+    latitude: 0,
+    longitude: 0,
+    pitch: 0,
+    zoom: 1
+  },
+  container: 'map'
+};
+
+/**
+ * A helper function to create a CARTO basemap on a 'map' DOM element, rendered using *Mapbox GL JS*
+ *
+ * Examples:
+ * ```javascript
+ *    // Several options to create the map are allowed
+ *    const deckMap = carto.viz.createMap();
+ *    const deckMap = carto.viz.createMap({ basemap: 'voyager'});
+ *    const deckMap = carto.viz.createMap({ basemap: 'voyager', view: { zoom: 4 } });
+ *    const deckMap = carto.viz.createMap({ basemap: 'positron', view: { zoom: 4, longitude: 3, latitude: 40, pitch: 45, bearing: 30 }, container: 'map' });
+ * ```
+ * @export
+ * @param {MapOptions} mapOptions
+ * @returns
+ */
+export function createMap(options: MapOptions = DEFAULT_OPTIONS) {
+  if (!window.deck.DeckGL) {
+    throw new CartoBaseMapError(
+      'Deck.gl library not found within window context. Please check documentation to know more about how to configure it.'
+    );
+  }
+
+  const chosenOptions = {
+    basemap: options.basemap || DEFAULT_OPTIONS.basemap || 'positron',
+    view: { ...DEFAULT_OPTIONS.view, ...options.view },
+    container: options.container || DEFAULT_OPTIONS.container
+  };
+
+  const deckMap = new (window.deck.DeckGL as any)({
```

Comment:
I've tried to fix this using the Deck.gl typings that we imported but it doesn't work because props are not updated. We might need to contribute to Deck.gl typings package.

If we wanted to fix this we would need to set our typings like this:
```js
interface Window {
    deck: {
      DeckGL: typeof Deck;
      GoogleMapsOverlay: GoogleMapsOverlay;
    };
}
```

and it would work. But let's leave it like this for now.

Reply:
I aggree.